echopype/echodata/echodata.py

```diff
@@ -374,10 +375,21 @@ def update_platform(self, extra_platform_data: xr.Dataset, time_dim="time"):
             extra_platform_data = extra_platform_data.drop_vars(trajectory_var)
             extra_platform_data = extra_platform_data.swap_dims({"obs": time_dim})
 
-        platform = self.platform.assign_coords(
-            time2=extra_platform_data[time_dim].values
+        extra_platform_data = extra_platform_data.sel(
+            {
+                time_dim: np.logical_and(
+                    self.beam["ping_time"].min() < extra_platform_data[time_dim],
+                    extra_platform_data[time_dim] < self.beam["ping_time"].max(),
+                )
+            }
+        )
+
+        platform = self.platform
+        platform = platform.drop_dims("location_time", errors="ignore")
+        platform = platform.assign_coords(
+            location_time=extra_platform_data[time_dim].values
         )
-        platform["time2"].attrs[
+        platform["location_time"].attrs[
             "long_name"
         ] = "time dimension for external platform data"
 
```

Comment:
I have a couple of comments about the `location_time` attribute assignments, going beyond `long_name`. Broadly, these should follow exactly what is done elsewhere when `location_time` is created during the conventional convert process (via open_raw):
- Exactly the [same attributes and attribute values as in ek60](https://github.com/OSOceanAcoustics/echopype/blob/dev/echopype/convert/set_groups_ek60.py#L206), including `long_name`
- End with a [`set_encodings` (or some variation of it) as done in ek60](https://github.com/OSOceanAcoustics/echopype/blob/dev/echopype/convert/set_groups_ek60.py#L321)

Currently, `long_name` is assigned a name that tries to inform users about the provenance of this time coordinate ("time dimension for external platform data"). I suggest that that provenance info be added instead via the [CF-recommended `history` attribute](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html#description-of-file-contents). Here's what the CF convention says about `history`:
> Provides an audit trail for modifications to the original data. Well-behaved generic netCDF filters will automatically append their name and the parameters with which they were invoked to the ~~global~~ history attribute of an input netCDF ~~file~~ (variable). We recommend that each line begin with a timestamp indicating the date and time of day that the program was executed.

I suggest something like this: "2021-11-08T11:13:00 Created from external platform data". Is the file name for the external platform data recorded in the provenance somewhere? It could also be added in the Platform group via a global `history` attribute.

cc @leewujung

Reply:
Ok, finally got to this!

@emiliom I agree with your suggestion on using the `history` attribute for recording the time this operation happens and adding the external platform filename.

> - Exactly the [same attributes and attribute values as in ek60](https://github.com/OSOceanAcoustics/echopype/blob/dev/echopype/convert/set_groups_ek60.py#L206), including `long_name`

For setting the attributes, perhaps using the same strategy as the one you did for the #480 ? 
This is nice, I feel we are now well on the way to address convention adherence issues! 😃 

> - End with a [`set_encodings` (or some variation of it) as done in ek60](https://github.com/OSOceanAcoustics/echopype/blob/dev/echopype/convert/set_groups_ek60.py#L321)

Agreed!

Reply:
Thanks! I'll start working on this later today